packages/cfd/src/Components/cfd-dxtrade-account-display.tsx

```diff
@@ -0,0 +1,156 @@
+
+type TStandPoint = {
+    financial_company: string;
+    gaming_company: string;
+    iom: boolean;
+    malta: boolean;
+    maltainvest: boolean;
+    svg: boolean;
+};
+
+type TOpenAccountTransferMeta = {
+    category: string;
+    type?: string;
+};
+
+type TCurrentList = DetailsOfEachMT5Loginid & {
+    enabled: number;
+};
+
+type TCFDDxtradeAccountDisplay = {
+    account_status?: object;
+    // TODO: update this type (DetailsOfEachMT5Loginid) when BE changed the schema
+    current_list: Record<string, TCurrentList>;
+    has_cfd_account_error: boolean;
+    has_real_account?: boolean;
+    is_accounts_switcher_on?: boolean;
+    is_demo_tab?: boolean;
+    is_eu_country?: boolean;
+    is_eu?: boolean;
+    is_loading?: boolean;
+    is_logged_in: boolean;
+    is_virtual?: boolean;
+    landing_companies?: LandingCompany;
+    onSelectAccount: (objCFDAccount: TObjectCFDAccount) => void;
+    openAccountTransfer: (
+        data: DetailsOfEachMT5Loginid | TTradingPlatformAccounts,
+        meta: TOpenAccountTransferMeta
+    ) => void;
+    openDerivRealAccountNeededModal?: () => void;
+    openPasswordManager: (login?: string, title?: string, group?: string, type?: string, server?: string) => void;
+    platform: TCFDPlatform;
+    should_enable_add_button?: boolean;
+    standpoint: TStandPoint;
+    toggleAccountsDialog?: (is_accounts_switcher_on?: boolean) => void;
+    toggleShouldShowRealAccountsList?: (is_should_show_real_acc_list?: boolean) => void;
+};
+
+const CFDDxtradeAccountDisplay = ({
+    current_list,
+    has_cfd_account_error,
+    has_real_account,
+    is_accounts_switcher_on,
+    is_demo_tab = false,
+    is_eu_country,
+    is_eu,
+    is_loading,
+    is_logged_in,
+    is_virtual,
+    onSelectAccount,
+    openAccountTransfer,
+    openDerivRealAccountNeededModal,
+    openPasswordManager,
+    platform,
+    should_enable_add_button,
+    standpoint,
+    toggleAccountsDialog,
+    toggleShouldShowRealAccountsList,
+}: TCFDDxtradeAccountDisplay) => {
+    const type = 'all';
+    const acc_type = 'dxtrade';
+    const category = is_demo_tab ? 'demo' : 'real';
+    const is_eu_user = (is_logged_in && is_eu) || (!is_logged_in && is_eu_country);
+    const current_list_index =
+        Object.keys(current_list).find(key => key.startsWith(`${platform}.${category}.${acc_type}`)) || '';
+
+    const existing_accounts_data = () => {
+        const acc = Object.keys(current_list).some(
+            key => key.startsWith(`${platform}.${category}.${acc_type}`) && current_list[key].enabled === 1
+        )
+            ? Object.keys(current_list)
+                  .filter(key => key.startsWith(`${platform}.${category}.${acc_type}`))
+                  .reduce((_acc, cur) => {
+                      _acc.push(current_list[cur]);
+                      return _acc;
+                  }, [] as DetailsOfEachMT5Loginid[])
+            : undefined;
+        return acc;
```

Comment:
```suggestion
        const keyword = `${platform}.${category}.${acc_type}`;
        const keys = Object.keys(current_list).filter(key => key.startsWith(keyword));
        const is_enabled = keys.some(key => current_list[key].enabled === 1);

        if (!is_enabled) return undefined;

        return keys.reduce<DetailsOfEachMT5Loginid[]>((accounts, current) => {
            accounts.push(current_list[current]);

            return accounts;
        }, []);
```
@hamid-deriv Can be simplified by using helper variables, Just a suggestion 🙇🏻
Also don’t trust my code it’s just an example 😅

Reply:
Changed.